docs/elasticsearch/index-templates.md

```diff
@@ -0,0 +1,204 @@
+
+---
+
+#### Table of contents
+1. TOC
+{:toc}
+
+
+---
+
+## Create template
+
+To create an index template, use a POST request:
+
+```json
+POST _template
+```
+
+This command creates a template named `daily_logs` and applies it to any new index that matches the regular expression `logs-2020-01-*` and also adds it to the `my_logs` alias:
+
+```json
+PUT _template/daily_logs
+{
+  "aliases": {
+    "my_logs": {}
+  },
+  "index_patterns": [
+    "logs-2020-01-*"
+  ],
+  "settings": {
+    "number_of_shards": 2,
+    "number_of_replicas": 1
+  },
+  "mappings": {
+    "properties": {
+      "timestamp": {
+        "type": "date",
+        "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+      },
+      "value": {
+        "type": "double"
+      }
+    }
+  }
+}
+```
+
+You should see the following response:
+
+```json
+{
+  "acknowledged": true
+}
+```
+
+Now, if you create an index named `logs-2020-01-01`:
+
+```json
+PUT logs-2020-01-01
+```
+
+It will have the mappings and settings defined in the `daily_logs` template:
+
+```json
+GET logs-2020-01-01
+```
+
+```json
+{
+  "logs-2020-01-01": {
+    "aliases": {
+      "my_logs": {}
+    },
+    "mappings": {
+      "properties": {
+        "timestamp": {
+          "type": "date",
+          "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+        },
+        "value": {
+          "type": "double"
+        }
+      }
+    },
+    "settings": {
+      "index": {
+        "creation_date": "1578107970779",
+        "number_of_shards": "2",
+        "number_of_replicas": "1",
+        "uuid": "U1vMDMOHSAuS2IzPcPHpOA",
+        "version": {
+          "created": "7010199"
+        },
+        "provided_name": "logs-2020-01-01"
+      }
+    }
+  }
+}
+```
+
+You can keep creating indexes: `logs-2020-01-02`, `logs-2020-01-03`, and so on, and they will all have the same mappings and settings.
+
+## Retrieve template
+
+To list all index templates:
+
+```json
+GET _cat/templates
+```
+
+To find a template by its name:
+
+```json
+GET _template/daily_logs
+```
+
+To get a list of all your templates:
+
+```json
+GET _template/daily_logs
+```
+
+To get a list of all templates that match a pattern:
+
+```json
+GET _template/daily*
+```
+
+To check if a particular template exists:
+
+```json
+HEAD _template/<name>
+```
+
+## Configure multiple templates
+
+You can create multiple index templates for your indices. If an index matches more than one template, all the mappings and settings are merged for that index.
+
+The values from the more recently created templates override the later ones. So you can define a few common settings in a generic template that can act as a catchall and then keep adding more specialized ones.
+
+You can also explicitly specify the priority of the templates using the `order` parameter. A template with a lower order number is applied first and a one with a higher order number overrides it.
```

Comment:
This feels like the method we should recommend. The idea that creation time of the template matters at all is kind of mind-boggling.

An even better approach is to explicitly specify template priority using the `order` parameter. Elasticsearch applies templates with lower order numbers first and then overrides them with templates that have higher order numbers.